File: sdks/typescript/src/resources/intelligence.ts
/** Memory self-management passes (POST /v1/intelligence/{decay,dedup,consolidate}). */

import type { HttpClient } from "../http.js";
import type { ConsolidateResult, DecayResult, DedupResult } from "../types.js";

export class IntelligenceResource {
  constructor(private readonly http: HttpClient) {}

  decay(namespace: string, threshold?: number): Promise<DecayResult> {
    const body: Record<string, unknown> = { namespace };
    if (threshold !== undefined) body.threshold = threshold;
    return this.http.request<DecayResult>("POST", "/v1/intelligence/decay", { body });
  }

  dedup(namespace: string, threshold?: number): Promise<DedupResult> {
    const body: Record<string, unknown> = { namespace };
    if (threshold !== undefined) body.threshold = threshold;
    return this.http.request<DedupResult>("POST", "/v1/intelligence/dedup", { body });
  }

  consolidate(
    namespace: string,
    sourceIds: string[],
    summary?: string,
  ): Promise<ConsolidateResult> {
    const body: Record<string, unknown> = { namespace, source_ids: sourceIds };
    if (summary !== undefined) body.summary = summary;
    return this.http.request<ConsolidateResult>("POST", "/v1/intelligence/consolidate", { body });
  }
}

File: sdks/python/src/scp_memory_sdk/resources/intelligence.py
"""Memory self-management passes (POST /v1/intelligence/{decay,dedup,consolidate})."""

from __future__ import annotations

from scp_memory_sdk._http import HttpClient
from scp_memory_sdk.models import ConsolidateResult, DecayResult, DedupResult


class IntelligenceResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def decay(self, *, namespace: str, threshold: float | None = None) -> DecayResult:
        body: dict = {"namespace": namespace}
        if threshold is not None:
            body["threshold"] = threshold
        return DecayResult.from_dict(
            self._http.request("POST", "/v1/intelligence/decay", json=body)
        )

    def dedup(self, *, namespace: str, threshold: float | None = None) -> DedupResult:
        body: dict = {"namespace": namespace}
        if threshold is not None:
            body["threshold"] = threshold
        return DedupResult.from_dict(
            self._http.request("POST", "/v1/intelligence/dedup", json=body)
        )

    def consolidate(
        self,
        *,
        namespace: str,
        source_ids: list[str],
        summary: str | None = None,
    ) -> ConsolidateResult:
        body: dict = {"namespace": namespace, "source_ids": source_ids}
        if summary is not None:
            body["summary"] = summary
        return ConsolidateResult.from_dict(
            self._http.request("POST", "/v1/intelligence/consolidate", json=body)
        )
